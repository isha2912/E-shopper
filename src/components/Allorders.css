.order-container {
  display: flex;
  flex-direction: column;
  margin: 3rem;
}
.top-table {
  display: flex;
  flex-direction: column;
}

.tab-header {
  display: flex;
  justify-content: space-between;
  background-color: black;
  color: white;
  height: 5em;
  width: 80rem;
}
.tab-data {
  display: flex;
  justify-content: space-between;
  height: 5em;
  width: 80rem;
}
.table2-top-header > th {
  background-color: blue;
}
