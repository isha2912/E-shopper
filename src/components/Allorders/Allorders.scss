.all-orders {
  display: flex;
  flex-direction: column;
  margin: 3rem;
  flex-wrap: wrap;
}
.top-table {
  display: flex;
  flex-direction: column;
  margin-top: 1rem;

  height: 20%;
  width: 20rem;
  border-collapse: collapse;
}

.tab-header {
  display: flex;
  justify-content: space-between;
  background-color: black;
  color: white;
  height: 5em;
  width: 80rem;
}
.tab-data {
  display: flex;
  justify-content: space-between;
  height: 5em;
  width: 80rem;
}
.table2-top-header > th {
  background-color: blue;
}
