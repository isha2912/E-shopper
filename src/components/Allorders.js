/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React from 'react';
import './Allorders.scss';

const Allorders = (props) => {
  const { allOrders } = props;
  return (
    <div className="order-container">
      <h1> Post Orders </h1>
      <div className="main-content">
        <table>
          <tr>
            <td>Order id</td>
            <td>Date</td>
            <td>Items</td>
          </tr>
          {allOrders.map((eachOrder) => (
            <tr key={eachOrder.id}>
              <td>{eachOrder.id}</td>
              <td>{new Date(eachOrder.date).toString()}</td>
              <td>
                {(eachOrder.items).map((eachItem) => (
                  <table key={eachItem.id}>
                    <tr>
                      <td>{eachItem.id}</td>
                      <td>{eachItem.name}</td>
                      <td>{eachItem.count}</td>
                    </tr>
                  </table>
                ))}
              </td>
            </tr>
          ))}
        </table>
      </div>

    </div>

  );
};
export default Allorders;

Allorders.propTypes = {

};
