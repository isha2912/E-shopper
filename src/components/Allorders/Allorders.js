/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React from 'react';
import './Allorders.scss';
import Table from '../Table/Table';
import filterUtil from '../../utils/api';

const Allorders = (props) => {
  const { allOrders } = props;
  const totalAmount = allOrders.map((eachOrder) => {
    const amount = eachOrder.items.reduce((acc, cur) => (acc + cur.price), 0);
    return amount;
  });
  return (
    <div className="all-orders">
      Post Orders
      <table className="top-table">
        <tbody>
          <tr>
            <th>Order id</th>
            <th> Date</th>
            <th>amount</th>
            <th> Items</th>
          </tr>
          {allOrders.map((eachOrder, index) => (
            <React.Fragment key={eachOrder.id}>
              <tr>
                <td>{eachOrder.id}</td>
                <td>{eachOrder.date.toString()}</td>
                <td>{totalAmount[index]}</td>
                <td>{eachOrder.items.length}</td>
              </tr>
              <tr>
                <Table cartItems={filterUtil.groupByCategory(eachOrder.items)} className="table1" />
              </tr>
            </React.Fragment>
          ))}
        </tbody>
      </table>
    </div>
  );
};
export default Allorders;

Allorders.propTypes = {

};
