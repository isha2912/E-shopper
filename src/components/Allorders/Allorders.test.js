import React from 'react';
import { screen, render } from '@testing-library/react';

import Allorders from './Allorders';

describe(Allorders.name, () => {
  const mockProps = {
    allOrders: [
      {
        items: [
          {
            id: 1,
            name: 'pp',
            count: 20,
            price: 12,
            category: 'idk',
          },
        ],
        id: 1,
        date: 123,
      },
    ],

  };
  it('should match the snapShot', () => {
    const { container } = render(<Allorders {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should display Past Orders and headers', () => {
    render(<Allorders {...mockProps} />);
    screen.getByText('Post Orders');
    screen.getByText('Order id');
    screen.getByText('Items');
  });
});
