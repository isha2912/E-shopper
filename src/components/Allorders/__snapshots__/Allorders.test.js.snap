// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allorders should match the snapShot 1`] = `
<div>
  <div
    class="all-orders"
  >
    Post Orders
    <table
      class="top-table"
    >
      <tbody>
        <tr>
          <th>
            Order id
          </th>
          <th>
             Date
          </th>
          <th>
            amount
          </th>
          <th>
             Items
          </th>
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            123
          </td>
          <td>
            12
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <div
            class="tab"
          >
            <table
              class="table1"
            >
              <thead>
                <tr
                  class="table1"
                >
                  <th>
                     ITEM DESCRIPTION 
                  </th>
                  <th>
                     UNIT PRICE
                  </th>
                  <th>
                     QUANTITY 
                  </th>
                  <th>
                     SUBTOTAL 
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    idk
                  </td>
                </tr>
                <tr>
                  <td>
                    pp
                  </td>
                  <td>
                    12
                  </td>
                  <td>
                    0
                  </td>
                  <td>
                    0
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
